code/gui/griddisplay.h

```diff
@@ -78,13 +116,15 @@ protected Q_SLOTS:
     bool m_start_pos_selected = false;
     bool m_end_pos_selected = false;
 
-    struct Coord {
-        int x;
-        int y;
-    };
-
     Coord m_start_position;
     Coord m_end_position;
+
+    Coord m_mouse_click_start;
+    Coord m_mouse_click_release;
+
+    QPoint select_start;
+    QRect select_box;
+    QRubberBand *rubber_band;
```

Comment:
if only a pointer is needed then you should do a forward class declaration instead of including the header